`test_collection_option_repeats` fails on ARM Linux

Feel free to close this, as it is probably a bug in Zsh, but I still wanted to report it

```py
________________________ test_collection_option_repeats ________________________

zsh_tester = <function zsh_tester.<locals>._make_tester at 0xfffff2fdb600>

    def test_collection_option_repeats(zsh_tester):
        """A ``list[Path]`` keyword option must accept repeated ``--file``.
    
        Without ``*`` prefix on the spec, zsh would refuse to complete a second
        ``--file`` value, breaking the natural ``--file a --file b --file c``
        usage of collection-typed parameters.
        """
        import os
        import tempfile
        from pathlib import Path as _Path
    
        app = App(name="files")
    
        @app.default
        def m(file: Annotated[list[_Path], Parameter(help="files")] = []):  # noqa: B006
            """Files."""
    
        tester = zsh_tester(app, "files")
        with tempfile.TemporaryDirectory() as td:
            (_Path(td) / "first.txt").write_text("x")
            cwd = _Path.cwd()
            try:
                os.chdir(td)
>               completions = tester.get_completions("files --file first.txt --file ")
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/build/source/tests/completion/test_zsh.py:1187: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/build/source/tests/completion/conftest.py:347: in get_completions
    child.expect("ZTEST> ", timeout=1.5)
/nix/store/9ccpx4wji5vh733zamz6lhxvplz05kx1-python3.13-pexpect-4.9.0/lib/python3.13/site-packages/pexpect/spawnbase.py:354: in expect
    return self.expect_list(compiled_pattern_list,
/nix/store/9ccpx4wji5vh733zamz6lhxvplz05kx1-python3.13-pexpect-4.9.0/lib/python3.13/site-packages/pexpect/spawnbase.py:383: in expect_list
    return exp.expect_loop(timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^
/nix/store/9ccpx4wji5vh733zamz6lhxvplz05kx1-python3.13-pexpect-4.9.0/lib/python3.13/site-packages/pexpect/expect.py:179: in expect_loop
    return self.eof(e)
           ^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pexpect.expect.Expecter object at 0xfffff2f923c0>
err = EOF('End Of File (EOF). Exception style platform.')

    def eof(self, err=None):
        spawn = self.spawn
    
        spawn.before = spawn._before.getvalue()
        spawn._buffer = spawn.buffer_type()
        spawn._before = spawn.buffer_type()
        spawn.after = EOF
        index = self.searcher.eof_index
        if index >= 0:
            spawn.match = EOF
            spawn.match_index = index
            return index
        else:
            spawn.match = None
            spawn.match_index = None
            msg = str(spawn)
            msg += '\nsearcher: %s' % self.searcher
            if err is not None:
                msg = str(err) + '\n' + msg
    
            exc = EOF(msg)
            exc.__cause__ = None # in Python 3.x we can use "raise exc from None"
>           raise exc
E           pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.
E           <pexpect.pty_spawn.spawn object at 0xfffff3163250>
E           command: /nix/store/8kd19j6n9qh9iq89d7d0xdvnsfjam1zw-zsh-5.9/bin/zsh
E           args: [b'/nix/store/8kd19j6n9qh9iq89d7d0xdvnsfjam1zw-zsh-5.9/bin/zsh', b'-i']
E           buffer (last 100 chars): ''
E           before (last 100 chars): '\x1b[?2004hfiles --file first.txt --file '
E           after: <class 'pexpect.exceptions.EOF'>
E           match: None
E           match_index: None
E           exitstatus: None
E           flag_eof: True
E           pid: 1249
E           child_fd: 13
E           closed: False
E           timeout: 5
E           delimiter: <class 'pexpect.exceptions.EOF'>
E           logfile: None
E           logfile_read: None
E           logfile_send: None
E           maxread: 2000
E           ignorecase: False
E           searchwindowsize: None
E           delaybeforesend: 0.05
E           delayafterclose: 0.1
E           delayafterterminate: 0.1
E           searcher: searcher_re:
E               0: re.compile('ZTEST> ')

/nix/store/9ccpx4wji5vh733zamz6lhxvplz05kx1-python3.13-pexpect-4.9.0/lib/python3.13/site-packages/pexpect/expect.py:122: EOF
--------------------------- snapshot report summary ----------------------------
24 snapshots passed.
=========================== short test summary info ============================
FAILED tests/completion/test_zsh.py::test_collection_option_repeats - pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.
========== 1 failed, 2375 passed, 6 skipped, 16 deselected in 42.86s ===========
```

Yes, for some reason, I can still see this bug

```
pytest flags: -m pytest -k not\ \(build_succeeds\)
============================= test session starts ==============================
platform linux -- Python 3.13.13, pytest-9.0.3, pluggy-1.6.0
rootdir: /build/source
configfile: pyproject.toml
plugins: cov-nixpkgs-stub-1.0.0, mock-3.15.1, syrupy-5.1.0
collected 2398 items / 12 deselected / 2386 selected                           

tests/apps/test_burgery.py ....                                          [  0%]
tests/cli/test_complete.py ........                                      [  0%]
tests/cli/test_generate_docs.py ...........                              [  0%]
tests/cli/test_run.py ...............                                    [  1%]
tests/completion/test_bash.py .......................................... [  3%]
..........                                                               [  3%]
tests/completion/test_behavior.py ...................................... [  5%]
................                                                         [  6%]
tests/completion/test_detect.py ......................                   [  6%]
tests/completion/test_fish.py .................................          [  8%]
tests/completion/test_install.py ..........................              [  9%]
tests/completion/test_zsh.py ........................................... [ 11%]
............F...                                                         [ 11%]
tests/config/test_common.py .............                                [ 12%]
tests/config/test_dict.py .............................................. [ 14%]
...                                                                      [ 14%]
tests/config/test_end2end.py ...                                         [ 14%]
tests/config/test_env.py ......                                          [ 14%]
tests/config/test_json.py ....                                           [ 15%]
tests/config/test_source_parameter.py ...............                    [ 15%]
tests/config/test_toml.py ..                                             [ 15%]
tests/config/test_toml_dataclass_list.py ........                        [ 16%]
tests/config/test_yaml.py .                                              [ 16%]
tests/config/test_yaml_dataclass_list.py .......                         [ 16%]
tests/py312/test_stdio_path.py ......................................    [ 18%]
tests/py312/test_type_alias_parameter.py ...........                     [ 18%]
tests/py312/test_type_alias_type.py ..........                           [ 18%]
tests/test_annotations.py ....................................           [ 20%]
tests/test_app_attribute_inheritance.py ...........                      [ 20%]
tests/test_app_name_derivation.py ..                                     [ 20%]
tests/test_app_stack_overrides.py .....                                  [ 21%]
tests/test_app_utils.py .....s                                           [ 21%]
tests/test_argument.py ................................................. [ 23%]
........                                                                 [ 23%]
tests/test_async.py ...........                                          [ 24%]
tests/test_bind_attrs.py ........                                        [ 24%]
tests/test_bind_basic.py ............................................... [ 26%]
........................................................................ [ 29%]
.................                                                        [ 30%]
tests/test_bind_boolean_flag.py ........................................ [ 31%]
..                                                                       [ 32%]
tests/test_bind_converter_validator.py ..........................        [ 33%]
tests/test_bind_custom_type.py ..                                        [ 33%]
tests/test_bind_dataclasses.py ...................s......                [ 34%]
tests/test_bind_dict.py ......                                           [ 34%]
tests/test_bind_empty_iterable.py ...............................        [ 35%]
tests/test_bind_env_var.py .....                                         [ 36%]
tests/test_bind_generic_class.py ......                                  [ 36%]
tests/test_bind_kwargs.py ...                                            [ 36%]
tests/test_bind_list.py ................................................ [ 38%]
                                                                         [ 38%]
tests/test_bind_namedtuple.py ...                                        [ 38%]
tests/test_bind_no_parse.py ...........                                  [ 39%]
tests/test_bind_none.py .                                                [ 39%]
tests/test_bind_pos_only.py ...........                                  [ 39%]
tests/test_bind_tuple.py .................                               [ 40%]
tests/test_bind_typed_dict.py ..........                                 [ 40%]
tests/test_bind_union.py .................                               [ 41%]
tests/test_bind_unpack_typed_dict.py ..........                          [ 41%]
tests/test_bind_var_pos.py ..                                            [ 41%]
tests/test_coercion.py ................................................. [ 43%]
......................................................                   [ 46%]
tests/test_command_collision.py ...                                      [ 46%]
tests/test_console.py .........                                          [ 46%]
tests/test_docs_base.py .....                                            [ 46%]
tests/test_docs_e2e.py ................                                  [ 47%]
tests/test_docs_snapshots.py ........................                    [ 48%]
tests/test_docs_types.py ........                                        [ 48%]
tests/test_edit.py ...........                                           [ 49%]
tests/test_enum_flag.py ..................                               [ 50%]
tests/test_env_var.py ........                                           [ 50%]
tests/test_error_console.py ............                                 [ 51%]
tests/test_error_formatter.py ......                                     [ 51%]
tests/test_exceptions.py ..................                              [ 52%]
tests/test_exceptions_rich.py .......................................... [ 53%]
......                                                                   [ 54%]
tests/test_future_annotations.py .......                                 [ 54%]
tests/test_fuzzy_command_matching.py ...................                 [ 55%]
tests/test_generate_docs.py ...............................              [ 56%]
tests/test_generate_html_docs.py ...............                         [ 57%]
tests/test_generate_rst_docs.py ...............                          [ 57%]
tests/test_group.py ...............                                      [ 58%]
tests/test_group_extractors.py .........                                 [ 58%]
tests/test_help.py ..................................................... [ 60%]
........................................................................ [ 63%]
....                                                                     [ 64%]
tests/test_help_customization.py ................................        [ 65%]
tests/test_html_formatter.py ...............                             [ 66%]
tests/test_instantiating_module.py ...                                   [ 66%]
tests/test_interactive_shell.py ...........                              [ 66%]
tests/test_issue_643_flag_change_bug.py ...........                      [ 67%]
tests/test_json_string_dict.py .......                                   [ 67%]
tests/test_json_string_list.py ......................................... [ 69%]
........................                                                 [ 70%]
tests/test_json_string_special_types.py ...................              [ 70%]
tests/test_lazy_commands.py ........................................     [ 72%]
tests/test_loader.py ...........ssss........                             [ 73%]
tests/test_markdown_formatter.py ...................                     [ 74%]
tests/test_meta.py ................                                      [ 75%]
tests/test_mkdocs_ext.py .....................................           [ 76%]
tests/test_name_transform.py .................                           [ 77%]
tests/test_nested_command_filtering.py ..........                        [ 77%]
tests/test_new_type.py ........                                          [ 78%]
tests/test_panel.py ...                                                  [ 78%]
tests/test_parameter.py .........................................        [ 79%]
tests/test_parameter_allow_leading_hyphen.py ..                          [ 79%]
tests/test_parameter_count.py ...................                        [ 80%]
tests/test_parameter_decorator.py .......                                [ 81%]
tests/test_parameter_n_tokens.py ..........................              [ 82%]
tests/test_performance.py .....                                          [ 82%]
tests/test_pydantic.py ...............................                   [ 83%]
tests/test_requires_equals.py ................                           [ 84%]
tests/test_result_action.py ............................................ [ 86%]
.....................................                                    [ 87%]
tests/test_rst_preprocessor.py ...............                           [ 88%]
tests/test_run.py .                                                      [ 88%]
tests/test_runtime_exceptions.py .........                               [ 88%]
tests/test_sphinx_ext.py .......................                         [ 89%]
tests/test_structured_type_parameter_kinds.py ..........                 [ 90%]
tests/test_subapp.py ........                                            [ 90%]
tests/test_subapp_flatten.py .................                           [ 91%]
tests/test_trio.py ...                                                   [ 91%]
tests/test_utils.py ...................................                  [ 92%]
tests/test_validate_command.py .                                         [ 92%]
tests/test_version.py .................                                  [ 93%]
tests/test_version_parameter.py ....                                     [ 93%]
tests/types/test_types_json.py .                                         [ 93%]
tests/types/test_types_number.py ...........                             [ 94%]
tests/types/test_types_path.py ......................................... [ 95%]
..................................................                       [ 98%]
tests/types/test_types_web.py ...........                                [ 98%]
tests/validators/test_validator_group.py ..........                      [ 98%]
tests/validators/test_validator_number.py ........                       [ 99%]
tests/validators/test_validator_path.py ..................               [100%]

=================================== FAILURES ===================================
________________________ test_collection_option_repeats ________________________

zsh_tester = <function zsh_tester.<locals>._make_tester at 0xfffff2fcff60>

    def test_collection_option_repeats(zsh_tester):
        """A ``list[Path]`` keyword option must accept repeated ``--file``.
    
        Without ``*`` prefix on the spec, zsh would refuse to complete a second
        ``--file`` value, breaking the natural ``--file a --file b --file c``
        usage of collection-typed parameters.
        """
        import os
        import tempfile
        from pathlib import Path as _Path
    
        app = App(name="files")
    
        @app.default
        def m(file: Annotated[list[_Path], Parameter(help="files")] = []):  # noqa: B006
            """Files."""
    
        tester = zsh_tester(app, "files")
        with tempfile.TemporaryDirectory() as td:
            (_Path(td) / "first.txt").write_text("x")
            cwd = _Path.cwd()
            try:
                os.chdir(td)
>               completions = tester.get_completions("files --file first.txt --file ")
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/build/source/tests/completion/test_zsh.py:1187: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/build/source/tests/completion/conftest.py:363: in get_completions
    child.expect("ZTEST> ", timeout=1.5)
/nix/store/9ccpx4wji5vh733zamz6lhxvplz05kx1-python3.13-pexpect-4.9.0/lib/python3.13/site-packages/pexpect/spawnbase.py:354: in expect
    return self.expect_list(compiled_pattern_list,
/nix/store/9ccpx4wji5vh733zamz6lhxvplz05kx1-python3.13-pexpect-4.9.0/lib/python3.13/site-packages/pexpect/spawnbase.py:383: in expect_list
    return exp.expect_loop(timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^
/nix/store/9ccpx4wji5vh733zamz6lhxvplz05kx1-python3.13-pexpect-4.9.0/lib/python3.13/site-packages/pexpect/expect.py:179: in expect_loop
    return self.eof(e)
           ^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pexpect.expect.Expecter object at 0xfffff31af230>
err = EOF('End Of File (EOF). Exception style platform.')

    def eof(self, err=None):
        spawn = self.spawn
    
        spawn.before = spawn._before.getvalue()
        spawn._buffer = spawn.buffer_type()
        spawn._before = spawn.buffer_type()
        spawn.after = EOF
        index = self.searcher.eof_index
        if index >= 0:
            spawn.match = EOF
            spawn.match_index = index
            return index
        else:
            spawn.match = None
            spawn.match_index = None
            msg = str(spawn)
            msg += '\nsearcher: %s' % self.searcher
            if err is not None:
                msg = str(err) + '\n' + msg
    
            exc = EOF(msg)
            exc.__cause__ = None # in Python 3.x we can use "raise exc from None"
>           raise exc
E           pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.
E           <pexpect.pty_spawn.spawn object at 0xfffff314fb10>
E           command: /nix/store/8kd19j6n9qh9iq89d7d0xdvnsfjam1zw-zsh-5.9/bin/zsh
E           args: [b'/nix/store/8kd19j6n9qh9iq89d7d0xdvnsfjam1zw-zsh-5.9/bin/zsh', b'--no-globalrcs', b'-i']
E           buffer (last 100 chars): ''
E           before (last 100 chars): '\x1b[?2004hfiles --file first.txt --file '
E           after: <class 'pexpect.exceptions.EOF'>
E           match: None
E           match_index: None
E           exitstatus: None
E           flag_eof: True
E           pid: 1296
E           child_fd: 13
E           closed: False
E           timeout: 5
E           delimiter: <class 'pexpect.exceptions.EOF'>
E           logfile: None
E           logfile_read: None
E           logfile_send: None
E           maxread: 2000
E           ignorecase: False
E           searchwindowsize: None
E           delaybeforesend: 0.05
E           delayafterclose: 0.1
E           delayafterterminate: 0.1
E           searcher: searcher_re:
E               0: re.compile('ZTEST> ')

/nix/store/9ccpx4wji5vh733zamz6lhxvplz05kx1-python3.13-pexpect-4.9.0/lib/python3.13/site-packages/pexpect/expect.py:122: EOF
--------------------------- snapshot report summary ----------------------------
24 snapshots passed.
=========================== short test summary info ============================
FAILED tests/completion/test_zsh.py::test_collection_option_repeats - pexpect.exceptions.EOF: End Of File (EOF). Exception style platform.
========== 1 failed, 2379 passed, 6 skipped, 12 deselected in 44.28s ===========
```

--- a/cyclopts/completion/install.py
+++ b/cyclopts/completion/install.py
@@ -62,14 +62,17 @@ def get_default_completion_path(shell: Literal["zsh", "bash", "fish"], prog_name
     """
     home = Path.home()
     if shell == "zsh":
+        # Namespace the file as ``_cyclopts_<prog>`` so the autoload entry
+        # doesn't shadow zsh helpers (``_files``, ``_directories``, etc.) when
+        # ``<prog>`` happens to match one.
         omz_completions = _detect_omz_completions_dir()
         if omz_completions is not None:
             omz_completions.mkdir(parents=True, exist_ok=True)
-            return omz_completions / f"_{prog_name}"
+            return omz_completions / f"_cyclopts_{prog_name}"
         # Vanilla zsh fallback
         zsh_completions = home / ".zsh" / "completions"
         zsh_completions.mkdir(parents=True, exist_ok=True)
-        return zsh_completions / f"_{prog_name}"
+        return zsh_completions / f"_cyclopts_{prog_name}"
     elif shell == "bash":
         bash_completions = home / ".local" / "share" / "bash-completion" / "completions"
         bash_completions.mkdir(parents=True, exist_ok=True)

--- a/cyclopts/completion/zsh.py
+++ b/cyclopts/completion/zsh.py
@@ -52,10 +52,15 @@ def generate_completion_script(app: "App", prog_name: str) -> str:
 
     completion_data = extract_completion_data(app)
 
+    # Namespace the function (and the install file) as ``_cyclopts_<prog>`` so
+    # that command names which happen to match a zsh completion helper —
+    # ``files``, ``directories``, ``describe``, etc. — don't shadow the builtin
+    # when compinit autoloads our script. Plain ``_<prog>`` would otherwise
+    # recurse on internal ``_files`` / ``_directories`` calls.
     lines = [
         f"#compdef {prog_name}",
         "",
-        f"_{prog_name}() {{",
+        f"_cyclopts_{prog_name}() {{",
         "  local line state",
         "",
     ]

--- a/cyclopts/core.py
+++ b/cyclopts/core.py
@@ -2462,7 +2462,7 @@ class App:
             Shell type for completion. If not specified, attempts to auto-detect current shell.
         output : Path | None
             Output path for the completion script. If not specified, uses shell-specific default:
-            - zsh: ~/.zsh/completions/_<prog_name> (or $ZSH_CUSTOM/completions/_<prog_name> with oh-my-zsh)
+            - zsh: ~/.zsh/completions/_cyclopts_<prog_name> (or $ZSH_CUSTOM/completions/_cyclopts_<prog_name> with oh-my-zsh)
             - bash: ~/.local/share/bash-completion/completions/<prog_name>
             - fish: ~/.config/fish/completions/<prog_name>.fish
         add_to_startup : bool

--- a/docs/source/shell_completion.rst
+++ b/docs/source/shell_completion.rst
@@ -104,7 +104,7 @@ For programmatic control, use :meth:`App.install_completion <cyclopts.App.instal
 Default Installation Paths
 ---------------------------
 
-- **Zsh**: ``~/.zsh/completions/_<app_name>``
+- **Zsh**: ``~/.zsh/completions/_cyclopts_<app_name>``
 - **Bash**: ``~/.local/share/bash-completion/completions/<app_name>``
 - **Fish**: ``~/.config/fish/completions/<app_name>.fish``
 

--- a/tests/completion/conftest.py
+++ b/tests/completion/conftest.py
@@ -294,7 +294,7 @@ class ZshCompletionTester(CompletionTesterBase):
 
     def validate_script_syntax(self) -> bool:
         with tempfile.TemporaryDirectory() as tmpdir:
-            comp_file = Path(tmpdir) / f"_{self.prog_name}"
+            comp_file = Path(tmpdir) / f"_cyclopts_{self.prog_name}"
             comp_file.write_text(self.completion_script)
             result = subprocess.run(
                 ["zsh", "-n", str(comp_file)],
@@ -312,7 +312,7 @@ class ZshCompletionTester(CompletionTesterBase):
 
         with tempfile.TemporaryDirectory() as tmpdir:
             td = Path(tmpdir)
-            (td / f"_{self.prog_name}").write_text(self.completion_script)
+            (td / f"_cyclopts_{self.prog_name}").write_text(self.completion_script)
 
             (td / ".zshrc").write_text(
                 "PROMPT='ZTEST> '\n"

--- a/tests/completion/test_install.py
+++ b/tests/completion/test_install.py
@@ -336,11 +336,11 @@ def test_install_completion_command_fish(temp_home, monkeypatch, capsys):
 
 
 def test_install_completion_zsh_ohmyzsh_default_path(omz_dir):
-    """Test that $ZSH set with valid dir installs to $ZSH/custom/completions/_testapp."""
+    """Test that $ZSH set with valid dir installs to $ZSH/custom/completions/_cyclopts_testapp."""
     app = App(name="testapp")
     install_path = app.install_completion(shell="zsh", add_to_startup=True)
 
-    assert install_path == omz_dir / "custom" / "completions" / "_testapp"
+    assert install_path == omz_dir / "custom" / "completions" / "_cyclopts_testapp"
     assert install_path.exists()
 
 
@@ -353,7 +353,7 @@ def test_install_completion_zsh_ohmyzsh_zsh_custom_env(omz_dir, temp_home, monke
     app = App(name="testapp")
     install_path = app.install_completion(shell="zsh", add_to_startup=True)
 
-    assert install_path == custom_dir / "completions" / "_testapp"
+    assert install_path == custom_dir / "completions" / "_cyclopts_testapp"
     assert install_path.exists()
 
 
@@ -373,7 +373,7 @@ def test_install_completion_zsh_ohmyzsh_dir_missing(temp_home, monkeypatch):
     app = App(name="testapp")
     install_path = app.install_completion(shell="zsh", add_to_startup=False)
 
-    expected = temp_home / ".zsh" / "completions" / "_testapp"
+    expected = temp_home / ".zsh" / "completions" / "_cyclopts_testapp"
     assert install_path == expected
     assert install_path.exists()
 

--- a/tests/completion/test_zsh.py
+++ b/tests/completion/test_zsh.py
@@ -1167,6 +1167,13 @@ def test_collection_option_repeats(zsh_tester):
     Without ``*`` prefix on the spec, zsh would refuse to complete a second
     ``--file`` value, breaking the natural ``--file a --file b --file c``
     usage of collection-typed parameters.
+
+    Doubles as a regression test for issue #821: prog_name ``files`` happens
+    to match zsh's built-in ``_files`` completion helper, so the generated
+    autoload entry must be namespaced (e.g. ``_cyclopts_files``) — otherwise
+    ``_arguments``'s ``:file:_files`` action recurses into our own function
+    and zsh hits its recursion limit (visible on macOS as a captured error
+    string, fatal pty EOF on aarch64 Linux/Nix).
     """
     import os
     import tempfile
@@ -1187,7 +1194,10 @@ def test_collection_option_repeats(zsh_tester):
             completions = tester.get_completions("files --file first.txt --file ")
         finally:
             os.chdir(cwd)
-    assert completions, f"expected file completion on second --file, got {completions!r}"
+    # Assert the actual file appears — not just *some* output. Without a
+    # tight assertion zsh's "recursion limit exceeded" error gets scraped
+    # as a "completion" and the test silently passes.
+    assert "first.txt" in completions, f"expected 'first.txt' in file completion on second --file, got {completions!r}"
 
 
 def test_multiple_iterable_positionals_emit_one_rest_spec(zsh_tester):

